plugins/mongo_db_plugin/include/eosio/mongo_db_plugin/bson.hpp

```diff
@@ -0,0 +1,248 @@
+/**
+ *   @file
+ *   @copyright defined in eos/LICENSE
+ */
+#pragma once
+
+#include <bsoncxx/builder/basic/document.hpp>
+#include <bsoncxx/builder/basic/kvp.hpp>
+#include <bsoncxx/builder/stream/document.hpp>
+#include <bsoncxx/document/value.hpp>
+#include <bsoncxx/exception/exception.hpp>
+#include <bsoncxx/string/to_string.hpp>
+#include <bsoncxx/types.hpp>
+
+#include <fc/variant.hpp>
+
+namespace eosio {
+void to_bson(const fc::variant_object& o, bsoncxx::builder::core& c);
+void to_bson(const fc::variants& v, bsoncxx::builder::core& c);
+void to_bson(const fc::variant& v, bsoncxx::builder::core& c);
+bsoncxx::document::value to_bson(const fc::variant& v);
+
+void from_bson(const bsoncxx::document::view& view, fc::mutable_variant_object& o);
+void from_bson(const bsoncxx::array::view& bson_array, fc::variants& a);
+template <typename T> void from_bson(const T& ele, fc::variant& v);
+fc::variant from_bson(const bsoncxx::document::view& view);
+} // namespace eosio
+
+namespace eosio {
+
+void to_bson(const fc::variant_object& o, bsoncxx::builder::core& c)
+{
+   auto itr = o.begin();
+   while (itr != o.end()) {
+      c.key_owned(itr->key());
+      to_bson(itr->value(), c);
+      ++itr;
+   }
+}
+
+void to_bson(const fc::variants& a, bsoncxx::builder::core& c)
+{
+   auto itr = a.begin();
+   while (itr != a.end()) {
+      to_bson(*itr, c);
+      ++itr;
+   }
+}
+
+void to_bson(const fc::variant& v, bsoncxx::builder::core& c)
+{
+   switch (v.get_type()) {
+      case fc::variant::null_type: {
+         c.append(bsoncxx::types::b_null{});
+         return;
+      }
+      case fc::variant::int64_type:
+      case fc::variant::uint64_type: {
+         c.append(v.as_int64());
+         return;
+      }
+      case fc::variant::double_type:
+         c.append(v.as_double());
+         return;
+      case fc::variant::bool_type:
+         c.append(v.as_bool());
+         return;
+      case fc::variant::string_type: {
+         c.append(v.as_string());
+         return;
+      }
+      case fc::variant::blob_type: {
+         bsoncxx::types::b_binary bin;
+         bin.sub_type = bsoncxx::binary_sub_type::k_binary;
+         bin.size = v.as_blob().data.size();
+         bin.bytes = reinterpret_cast<uint8_t*>(&(*v.as_blob().data.begin()));
+         c.append(bin);
+         return;
```

Comment:
Have you verified that hex strings are identical to the previous approach? For example the `hex_data` of an action?

Reply:
Yes, identical. hex_data is string type.